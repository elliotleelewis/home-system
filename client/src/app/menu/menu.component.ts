import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { RouterLink, RouterLinkActive } from '@angular/router';

import { LocationRef } from '../refs/location.ref';

@Component({
	selector: 'app-menu',
	templateUrl: './menu.component.html',
	changeDetection: ChangeDetectionStrategy.OnPush,
	imports: [RouterLink, RouterLinkActive],
})
export class MenuComponent {
	private readonly _locationRef = inject(LocationRef);

	get homebridgeUrl() {
		const url = new URL(this._locationRef.location.origin);
		url.port = '8581';
		return url.toString();
	}
}
