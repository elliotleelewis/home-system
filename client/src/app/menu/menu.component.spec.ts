import type { ComponentFixture } from '@angular/core/testing';
import { MockBuilder, MockRender } from 'ng-mocks';

import { MenuComponent } from './menu.component';

describe('MenuComponent', () => {
	let component: MenuComponent;
	let fixture: ComponentFixture<MenuComponent>;

	beforeEach(() => {
		return MockBuilder(MenuComponent);
	});

	beforeEach(() => {
		fixture = MockRender(MenuComponent);
		component = fixture.componentInstance;
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should render', () => {
		expect(fixture).toMatchSnapshot();
	});
});
