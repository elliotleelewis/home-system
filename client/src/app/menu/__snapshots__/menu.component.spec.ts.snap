// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuComponent should render 1`] = `
<mock-render>
  <app-menu>
    <div
      class="flex flex-col"
    >
      <a>
        <h1
          class="p-4 text-center text-3xl font-bold"
        >
          Home System
        </h1>
      </a>
      <ul
        class="menu rounded-box bg-base-200"
      >
        <div>
          Loading...
        </div>
      </ul>
    </div>
  </app-menu>
</mock-render>
`;
